### PR TITLE
Migrating to the PyData theme for the online docs

### PR DESCRIPTION
**Pull request by maximelucas**
This switches the theme of the onlines docs to the PyData theme.

Some reorganization is needed to fit with the theme.

### CODE REVIEW

**Comment by nwlandry** [comment]
I won't have the chance to do a thorough review by the time you need it, but overall looks great! Thanks for making this happen!

Quick things I noticed:
* where is the using XGI page?
* XGI-DATA table looks funky in dark mode
* "Copyright Copyright" in the footer

**Comment by maximelucas** [comment]
Thanks for the quick initial review Nich!

- "projects using XGI" page is linked to in text of the home page.
- good catch, I disabled the mode switch for now
- about the paths, I managed to shorted them for the functions, but not the module paths in the toc (it was like that in our current online docs already). I suggest we can try again in another PR

**Comment by maximelucas** [comment]
Summary of what I did for the record:
- New 1-level structure is now: Homepage, Installing, User Guides, API Reference, XGI-DATA, Gallery, and Development
- Homepage: is now more concise (I moved some sections in other places and linked them)
- Installing: concise but links to a "more options" page
- User Guides: new page, presenting the categories of tutorials
- API Ref: added a small description of the API at the top
- XGI-Data: added link to documentation (fix #548)
- Development: now combines "Contribute" and "Contributors"

I have ideas of options of things to add in the future, maybe a page for Releases, a better Gallery, etc.

File: docs/source/index.rst [hunk 1]
```diff
@@ -3,133 +3,74 @@
...
 
-   about
-   contributors
-   higher-order
+   installing
+   user_guides
+   reference
    xgi-data
    gallery
-   using-xgi
```
Comment (nwlandry) [inline docs/source/index.rst:15]: Can you restore using-xgi in the sidebar?

**Review by nwlandry — APPROVED** [review]

**Pull request merged**